client/components/map.jsx

```diff
@@ -40,13 +49,108 @@ class Mapbox extends React.Component {
         accessToken: mapboxgl.accessToken,
         mapboxgl: mapboxgl
       })
+        .on('result', result => {
+          this.setState({ inLocations: null, showFinish: false });
+          const locationName = result.result['place_name_en-US'];
+          if (!this.state.locations[0]) {
+            this.setState({
+              location: locationName,
+              inLocations: false
+            });
+          } else {
+            for (let i = 0; i < this.state.locations.length; i++) {
+              if (this.state.locations[i] === locationName) {
+                this.setState({
+                  location: locationName,
+                  inLocations: true
+                }, () => setTimeout(() => {
+                  this.setState({
+                    showFinish: true
+                  });
+                }, 2000));
+                break;
+              } else {
+                this.setState({
+                  location: locationName,
+                  inLocations: false
+                });
+              }
+            }
+          }
+        })
     );
   }
 
+  handleClick() {
+    if (!this.state.location) {
+      return;
+    }
+    if (!this.state.inLocations) {
+      const newLocation = this.state.location;
+      this.setState({
+        locations: this.state.locations.concat(newLocation),
+        inLocations: true,
+        showFinish: false
+      }, () => setTimeout(() => {
+        this.setState({
+          showFinish: true
+        });
+      }, 2000));
+    } else {
+      window.alert(`${this.state.location.split(',')[0]} has already been added.`);
+    }
+  }
+
+  setAddValue() {
+    if (!this.state.location) {
+      return 'Search for a location to add';
+    } else if (!this.state.inLocations || this.state.inLocations === null) {
+      return `Add ${this.state.location.split(',')[0]}`;
+    } else {
+      return `${this.state.location.split(',')[0]} Added!`;
+    }
+  }
```

Comment:
A method named `setBlahBlah` doesn't usually return something. Would the method name make more sense if it was `getButtonText()`?